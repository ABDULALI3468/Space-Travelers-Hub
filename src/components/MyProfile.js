import React from 'react';
import { useSelector } from 'react-redux';
import Table from 'react-bootstrap/Table';
import '../styles/MyProfile.css';

const MyProfile = () => {
  const reservedMissions = useSelector((state) => (
    state.missions.filter((mission) => mission.reserved)
  ));

  const reservedRockets = useSelector((state) => (
    state.rockets.filter((rocket) => rocket.reserved)
  ));

  return (
    <div className="profile-content">
      <section className="reserved-missions">
        <h1>My Missions</h1>
        <Table bordered>
          <tbody>
            {
                reservedMissions.map((mission) => (
                  <tr key={mission.mission_id}>
                    <td>{mission.mission_name}</td>
                  </tr>
                ))
              }
          </tbody>
        </Table>
      </section>

      <section className="reserved-rockets">
        <h1>My Rockets</h1>
        <Table bordered>
          <tbody>
            {
              reservedRockets.map((rocket) => (
                <tr key={rocket.id}>
                  <td>{rocket.name}</td>
                </tr>
              ))
          }
          </tbody>
        </Table>
      </section>
    </div>
  );
};

export default MyProfile;
