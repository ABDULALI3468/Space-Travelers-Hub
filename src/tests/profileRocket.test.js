import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MyProfile from '../components/MyProfile';
import store from '../Redux/ConfigureStore';

it('Check if it renders correctly', () => {
  const output = renderer
    .create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    )
    .toJSON();
  expect(output).toMatchSnapshot();
});
