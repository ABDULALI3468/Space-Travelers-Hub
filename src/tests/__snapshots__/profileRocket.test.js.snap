// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if it renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="rockets"
  >
    <h1
      className="rockets-heading"
    >
      Rockets
    </h1>
  </div>
</div>
`;
